controllers/iamrole_controller.go

```diff
@@ -229,29 +229,28 @@ func (r *IamroleReconciler) HandleReconcile(ctx context.Context, req ctrl.Reques
 		fallthrough
 	default:
 
-		resp, err := r.IAMClient.CreateRole(ctx, *input)
+		// Default behavior on new Iamrole resource state is to go off and create it
+		resp, err := r.IAMClient.EnsureRole(ctx, *input)
 		if err != nil {
 			log.Error(err, "error in creating a role")
 			state := iammanagerv1alpha1.Error
 
-			if strings.Contains(err.Error(), awsapi.RoleAlreadyExistsError) {
+			// This check verifies whether or not the IAM Role somehow already exists, but is allocated to another namespace based on the tag applied to it.
+			if strings.Contains(err.Error(), awsapi.RoleExistsAlreadyForOtherNamespace) {
```

Comment:
I renamed this exception string because it really wasn't checking if the role exists, but was checking if it exists and was tagged for another namespace.